lib/vector-buffers/src/topology/channel/sender.rs

```diff
@@ -59,16 +92,32 @@ where
                 let mut writer = writer.lock().await;
 
                 if let Err(e) = writer.write_record(item).await {
-                    // Can't really do much except panic here. :sweat:
-                    panic!("writer hit unrecoverable error during write: {}", e);
+                    // TODO: Actually handle recoverable errors when we have sink error handling
+                    // capabilities i.e. a record that's too big to encode/serialize could
+                    // theoretically be passed back and handled somewhere else... we don't have to
+                    // drop it.
+
+                    // We want to get right to the meat of things by emitting an error message here,
+                    // because otherwise the bubbled-up error will be a bit plain without more
+                    // context.
+                    //
+                    // TODO: Should we actually take the Go-style approach and just build new errors
+                    // that are chained together e.g. `disk buffer error: i/o error: no space left
+                    // on device`. That one would be more useful when emitted by the component
+                    // task's error handling code... but it's also not our typical pattern.
+                    if !e.is_recoverable() {
+                        error!("Disk buffer writer has encountered an unrecoverable error.");
+                    }
+
+                    return Err(SendError::new(e));
                 }
 
                 Ok(())
```

Comment:
This reads like a `match`, or even `.map_err` like the other paths.

Reply:
It could be a `map_err`, sure.

Reply:
Switched this to use `map_err` in ddd91a8d6.